pynuodb/result_set.py

```diff
@@ -32,11 +32,11 @@ def fetchone(self, session):
         """
         :type session EncodedSession
         """
+        if self.results_idx == len(self.results) and not self.complete:
+            session.fetch_result_set_next(self)
```

Comment:
Shouldnt it check if 'fetch_result_set_next' does actually fetch more rows and if so, return the first one? I mean, it was wrong to always assume it would get a new row, but isn't it also wrong to never assume that it would?

Reply:
that is what it does, isn't it? If fetch_result_set doesn't return anything the condition in the following IF is true, so we return None. If it does, then we return the first one.

Reply:
Sorry, I misread the patch and missed the fact that the if statement was still there on new line 38/old line 35. I thought it called 'fetch' and then immediately returned None. Thanks for clarifying.